configure.ac

```diff
@@ -409,6 +409,8 @@ _MY_CXX_CHECK_OPT(CFG_CXXFLAGS_SRC,-Qunused-arguments)
 _MY_CXX_CHECK_OPT(CFG_CXXFLAGS_SRC,-faligned-new)
 _MY_CXX_CHECK_OPT(CFG_CXXFLAGS_SRC,-Wno-unused-parameter)
 _MY_CXX_CHECK_OPT(CFG_CXXFLAGS_SRC,-Wno-shadow)
+_MY_CXX_CHECK_OPT(CFG_CXXFLAGS_SRC,-Wthread-safety)
+_MY_CXX_CHECK_OPT(CFG_CXXFLAGS_SRC,-DVL_THREADED)
```

Comment:
I think that configure/make changes should be a separate patch.  But
1. We should not be setting VL_THREADED based on a compiler switch.  I think all compilers we support have C++11 now, so we can always be setting VL_THREADED when compiling Verilator.
2. Thread-safety is already turned on in the WEXTRA flags, which is where it should be.